packages/ruleset/src/functions/well-defined-dictionaries.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2024 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const {
+  isObject,
+  isObjectSchema,
+  schemaHasConstraint,
+  validateNestedSchemas,
+} = require('@ibm-cloud/openapi-ruleset-utilities');
+const { LoggerFactory } = require('../utils');
+
+let ruleId;
+let logger;
+
+module.exports = function (schema, _opts, context) {
+  if (!logger) {
+    ruleId = context.rule.name;
+    logger = LoggerFactory.getInstance().getLogger(ruleId);
+  }
+
+  return validateNestedSchemas(schema, context.path, wellDefinedDictionaries);
+};
+
+function wellDefinedDictionaries(schema, path) {
+  // We only care about object schemas.
+  if (!isObject(schema) || !isObjectSchema(schema)) {
+    return [];
+  }
+
+  logger.debug(
+    `${ruleId}: checking object schema at location: ${path.join('.')}`
+  );
+
+  // Dictionaries should have additionalProperties defined on them.
+  // If the schema doesn't, make sure it has properties and then
+  // abandon the check.
+  if (!schemaDefinesField(schema, 'additionalProperties')) {
+    if (!schemaDefinesField(schema, 'properties')) {
+      return [
+        {
+          message:
+            'Object schemas must either define properties or well-defined additional properties',
```

Comment:
```suggestion
            'Object schemas must define either properties or additionalProperties',
```
Suggested wording change to make it clear. I'm not sure that "well-defined additional properties" is clear enough.